From 109aa15ad0fcdf7d021aae6cc052c0ea6e3d4a20 Mon Sep 17 00:00:00 2001
From: Anurag Das <anurdas@codeaurora.org>
Date: Mon, 18 Dec 2017 14:47:01 +0530
Subject: [PATCH 13/30] Wifi: Restore hidden flag for hidden networks on update
 network

Currently when user modifies any of the network parameter for hidden
networks, hidden flag is not updated. As a result this network is no
longer part of hidden networks, we are unable to reconnect when user
triggers wifi off and on.

This commit sets the value of hiddenSSID flag of config at the Settings
side on network update.
---
 src/com/android/settings/wifi/WifiConfigController.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/com/android/settings/wifi/WifiConfigController.java b/src/com/android/settings/wifi/WifiConfigController.java
index ecd4b5966e..d7bccc0d07 100644
--- a/src/com/android/settings/wifi/WifiConfigController.java
+++ b/src/com/android/settings/wifi/WifiConfigController.java
@@ -517,6 +517,7 @@ public class WifiConfigController implements TextWatcher,
                     mAccessPoint.getSsidStr());
         } else {
             config.networkId = mAccessPoint.getConfig().networkId;
+            config.hiddenSSID = mAccessPoint.getConfig().hiddenSSID;
         }
 
         config.shared = mSharedCheckBox.isChecked();
-- 
2.11.0

